src/emscripten-optimizer/simple_ast.h

```diff
@@ -920,22 +920,22 @@ struct JSPrinter {
         char *test = end;
         // remove zeros, and also doubles can use at most 24 digits, we can truncate any extras even if not zero
         while ((*test == '0' || test - buffer > 24) && test > buffer) test--;
-        int num = end - test;
+        size_t num = end - test;
         if (num >= 3) {
           test++;
           test[0] = 'e';
           if (num < 10) {
-            test[1] = '0' + num;
+            test[1] = (char)('0' + num);
```

Comment:
Maybe instead of casting everywhere here we should just introduce a char temp (convert num to a char once), and add that to `'0'`. We could also add an assert at the conversion time.